api/types.go

```diff
@@ -415,7 +422,7 @@ func (d *Duration) UnmarshalJSON(b []byte) (err error) {
 		if t < 0 {
 			d.Duration = time.Duration(math.MaxInt64)
 		} else {
-			d.Duration = time.Duration(t * float64(time.Second))
+			d.Duration = time.Duration(int(t) * int(time.Second))
```

Comment:
I switched this to just handle everything as an int. I don't think we need anything more granular than that.